### PR TITLE
Registry

### PR DESCRIPTION
**Pull request by cbizon**
Smart API registration rules

### CODE REVIEW

**Review by marcdubybroad — APPROVED** [review]

**Review by andrewsu — APPROVED** [review]

**Review by newgene — APPROVED** [review]

File: SmartAPIRegistration.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+"translator" must be included as a top-level OpenAPI tag.
+
+### x-translator element
+
+The registration must contain a compliant x-translator element as described [here](https://github.com/NCATSTranslator/translator_extensions/).  Among other things, the x-translator element defines the infores identifier for this component: an identifier that is common across environments such as dev, test, and production.
+
+### x-trapi element
+
+If the component implements a TRAPI interface, then the registration must contain a compliant x-trapi element as described [here](https://github.com/NCATSTranslator/translator_extensions/).
```
Comment (newgene) [inline SmartAPIRegistration.md:24]: also requires a `trapi` tag. And you may link to the specific x-trapi extension at:

https://github.com/NCATSTranslator/translator_extensions/tree/production/x-trapi

File: SmartAPIRegistration.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+## Requirements
+
+### Translator tag
+
+"translator" must be included as a top-level OpenAPI tag.
+
+### x-translator element
+
+The registration must contain a compliant x-translator element as described [here](https://github.com/NCATSTranslator/translator_extensions/).  Among other things, the x-translator element defines the infores identifier for this component: an identifier that is common across environments such as dev, test, and production.
```
Comment (newgene) [inline SmartAPIRegistration.md:20]: can use this more specific link for `x-translator`:

https://github.com/NCATSTranslator/translator_extensions/tree/production/x-translator

**Review by newgene — COMMENTED** [review]
I approved already, here are just two minor comments.

File: SmartAPIRegistration.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+## Requirements
+
+### Translator tag
+
+"translator" must be included as a top-level OpenAPI tag.
+
+### x-translator element
+
+The registration must contain a compliant x-translator element as described [here](https://github.com/NCATSTranslator/translator_extensions/tree/production/x-translator).  Among other things, the x-translator element defines the infores identifier for this component: an identifier that is common across environments such as dev, test, and production.
```
Comment (vdancik) [inline SmartAPIRegistration.md:20]: > . . . common across environments such as dev, test, and production

Should we use terms defined in `x-maturity` tag: "... common across environments such as development, testing, and production"?

File: SmartAPIRegistration.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+If the component implements a TRAPI interface, then "trapi" must be included as a top-level OpenAPI tag.
+
+### x-trapi element
+
+If the component implements a TRAPI interface, then the registration must contain a compliant x-trapi element as described [here](https://github.com/NCATSTranslator/translator_extensions/tree/production/x-trapi).
+
+### Uptime checking
+
+Registrations must enable uptime checking as described [here](https://smart-api.info/faq#api-monitor).
```
Comment (newgene) [inline SmartAPIRegistration.md:32]: here is a more detailed uptime documentation you can link to instead: https://github.com/SmartAPI/smartAPI/blob/master/docs/uptime.md

**Review by edeutsch — APPROVED** [review]

**Review by suihuang-ISB — APPROVED** [review]

**Comment by cbizon** [comment]
At the risk of rechurning things- I'm curious how people are handling their smartAPI registrations.  The method that's currently being employed across a number of ranking agent / exposures / SRI tools is to create the openapi spec within the instance based on the deployment metadata.

Because of that, the infores/trapi granularity is somewhat awkward to implement, and an infores/x-maturity level would be more natural.   Thoughts?

**Comment by edeutsch** [comment]
yes, it's a bit awkward, but it seems like the only sensible way to do it. I was just setting up our deployment as well. This is what it looks like now:
https://arax.ncats.io/api/arax/v1.2/ui/
image

Is this what everyone expects to see? Good?

**Comment by cbizon** [comment]
> yes, it's a bit awkward, but it seems like the only sensible way to do it.

I guess that's what I'm wondering... is this more sensible than one smartAPI registration per infores/x-maturity?    IMO they're equally reasonable, but (for my team at least) the infores/x-maturity one is more easily implemented.

**Comment by edeutsch** [comment]
> > yes, it's a bit awkward, but it seems like the only sensible way to do it.
> 
> I guess that's what I'm wondering... is this more sensible than one smartAPI registration per infores/x-maturity? IMO they're equally reasonable, but (for my team at least) the infores/x-maturity one is more easily implemented.

I am thinking the opposite. I don't think either option is more difficult to implement for us. However, I am thinking that the inflation of SmartAPI records by a factor of 3 or 4 by having registrations per infores/x-maturity seems more messy.

But I don't feel strongly.

**Pull request merged**